build.rs

```diff
@@ -489,6 +490,16 @@ fn build_library(
                 .and_then(|f| f.to_str())
                 .expect("No filename"),
         );
+
+        if &target.os == "wasi" {
+            let ranlib_cmd = std::env::var("RANLIB");
```

Comment:
Please write a comment explaining why this chunk of code is here.

Wouldn't we need to output a `cargo:rerun-if-env-changed=RANLIB` here?